pkg/proc/gdbserial/gdbserver.go

```diff
@@ -261,6 +262,18 @@ func (p *Process) Connect(conn net.Conn, path string, pid int) error {
 		}
 	}
 
+	if path == "" {
+		// try using jGetLoadedDynamicLibrariesInfos which is the only way to do
+		// this supported on debugserver (but only on macOS >= 12.10)
+		images, _ := p.conn.getLoadedDynamicLibraries()
```

Comment:
The problem is that the interface of jGetLoadedDynamicLibrariesInfo changed in a backwards incompatible way in some version of macOS (I don't remember which one). We are relying on the new interface but if we don't ignore the error we here we would have to filter out the error that the old jGetLoadedDynamicLibrariesInfo returns when you try to use it in the new way and I don't know what that is.

Ignoring the error should be fine here, we still don't have path which means we will fail later on because we can't open the executable (which could happen for other reasons as well).